packages/wonder-blocks-modal/src/components/flexible-panel.tsx

```diff
@@ -0,0 +1,231 @@
+    style?: StyleType;
+    /**
+     * Called when the close button is clicked.
+     *
+     * If you're using `ModalLauncher`, you should not use this prop!
+     * Instead, to listen for when the modal closes, add an `onClose` handler
+     * to the `ModalLauncher`.  Doing so will throw an error.
+     */
+    onClose?: () => unknown;
+    /**
+     * Test ID used for e2e testing.
+     *
+     * In this case, this `testId` comes from the `testId` prop defined in the
+     * Dialog variant (e.g. OnePaneDialog).
+     */
+    testId?: string;
+};
+
+/**
+ * FlexiblePanel is the content container.
+ *
+ * **Implementation notes:**
+ *
+ * If you are creating a custom Dialog, make sure to follow these guidelines:
+ * - Make sure to add this component inside the [ModalDialog](/#modaldialog).
+ * - If needed, you can also add a [FlexibleHeader](/#Flexibleheader) using the
+ *   `header` prop. Same goes for [FlexibleFooter](/#Flexiblefooter).
+ * - If you need to create e2e tests, make sure to pass a `testId` prop. This
+ *   will be passed down to this component using a sufix: e.g.
+ *   `some-random-id-FlexiblePanel`. This scope will be propagated to the
+ *   CloseButton element as well: e.g. `some-random-id-CloseButton`.
+ *
+ * ```js
+ * <ModalDialog>
+ *      <FlexiblePanel content={"custom content goes here"} />
+ * </ModalDialog>
+ * ```
+ */
+export default function FlexiblePanel({
+    backgroundStyles,
+    closeButtonVisible = true,
+    scrollOverflow = true,
+    content,
+    titleId,
+    title,
+    footer,
+    onClose,
+    style,
+    testId,
+}: Props) {
+    const renderMainContent = React.useCallback((): React.ReactNode => {
+        const contentNode =
+            typeof content === "function" ? (
+                content({title})
+            ) : (
+                <>
+                    {title}
+                    {content}
+                </>
+            );
+        const mainContent = ModalContent.isComponentOf(contentNode) ? (
+            (contentNode as React.ReactElement<PropsFor<typeof ModalContent>>)
+        ) : (
+            <ModalContent>{contentNode}</ModalContent>
+        );
+
+        if (!mainContent) {
+            return mainContent;
+        }
+
+        return React.cloneElement(mainContent, {
+            // Pass the scrollOverflow and header in to the main content
+            scrollOverflow,
+            // We override the styling of the main content to help position
+            // it if there is a footer or close button being
+            // shown. We have to do this here as the ModalContent doesn't
+            // know about things being positioned around it.
+            style: [!!footer && styles.hasFooter, mainContent.props.style],
+        });
+    }, [title, content, footer, scrollOverflow]);
+
+    const mainContent = renderMainContent();
+
+    const defaultBackgroundStyle = {
+        backgroundColor: semanticColor.surface.primary,
+    };
+
+    const combinedBackgroundStyles = {
```

Comment:
This is where a custom background image/color can be passed in to the component where it displays. It was hard-coded to `surface.primary` before and couldn't be overridden from up at the `FlexibleDialog` level.